cms/djangoapps/contentstore/utils.py

```diff
@@ -1688,6 +1691,136 @@ def get_course_videos_context(course_block, pagination_conf, course_key=None):
     return course_video_context
 
 
+def get_container_handler_context(request, usage_key):
+    """
+    Utils is used to get context for container xblock requests.
+    It is used for both DRF and django views.
+    """
+
+    from cms.djangoapps.contentstore.views.component import (
+        _get_item_in_course,
+        get_component_templates,
+        get_unit_tags,
+        CONTAINER_TEMPLATES,
+        LIBRARY_BLOCK_TYPES,
+    )
+    from cms.djangoapps.contentstore.helpers import get_parent_xblock, is_unit
+    from cms.djangoapps.contentstore.xblock_storage_handlers.view_handlers import (
+        add_container_page_publishing_info,
+        create_xblock_info,
+    )
+    from openedx.core.djangoapps.content_staging import api as content_staging_api
+
+    try:
+        course, xblock, lms_link, preview_lms_link = _get_item_in_course(request, usage_key)
+    except ItemNotFoundError:
+        return HttpResponseBadRequest()
+
+    component_templates = get_component_templates(course)
+    ancestor_xblocks = []
+    parent = get_parent_xblock(xblock)
+    action = request.GET.get('action', 'view')
+
+    is_unit_page = is_unit(xblock)
+    unit = xblock if is_unit_page else None
+
+    if is_unit_page and use_new_unit_page(course.id):
+        return redirect(get_unit_url(course.id, unit.location))
+
+    is_first = True
+    block = xblock
+
+    # Build the breadcrumbs and find the ``Unit`` ancestor
+    # if it is not the immediate parent.
+    while parent:
+
+        if unit is None and is_unit(block):
+            unit = block
+
+        # add all to nav except current xblock page
+        if xblock != block:
+            current_block = {
+                'title': block.display_name_with_default,
+                'children': parent.get_children(),
+                'is_last': is_first
+            }
+            is_first = False
+            ancestor_xblocks.append(current_block)
+
+        block = parent
+        parent = get_parent_xblock(parent)
+
+    ancestor_xblocks.reverse()
+
+    assert unit is not None, "Could not determine unit page"
+    subsection = get_parent_xblock(unit)
+    assert subsection is not None, "Could not determine parent subsection from unit " + str(
+        unit.location)
+    section = get_parent_xblock(subsection)
+    assert section is not None, "Could not determine ancestor section from unit " + str(unit.location)
```

Comment:
```suggestion
if unit is None:
    raise ValueError("Could not determine unit page")

subsection = get_parent_xblock(unit)
if subsection is None:
    raise ValueError(f"Could not determine parent subsection from unit {unit.location}")

section = get_parent_xblock(subsection)
if section is None:
    raise ValueError(f"Could not determine ancestor section from unit {unit.location}")
```

Is it better to replace with `raise` instead of assert since it is not a test function?

Reply:
I suggest leaving it as it was

Reply:
added new commit with this replacement